eras/conway/impl/testlib/Test/Cardano/Ledger/Conway/Imp/GovSpec.hs

```diff
@@ -1166,6 +1178,40 @@ networkIdWithdrawalsSpec =
               Testnet
         ]
 
+    it "Fails for empty withdrawals" $ do
+      rwdAccount1 <- freshKeyHash >>= getRewardAccountFor . KeyHashObj
+      rwdAccount2 <- freshKeyHash >>= getRewardAccountFor . KeyHashObj
+      ( \wdrl ->
+          submitFailingGovAction
+            wdrl
+            [injectFailure $ ZeroTreasuryWithdrawals wdrl]
+        )
+        (TreasuryWithdrawals Map.empty SNothing)
```

Comment:
This is an interesting way to avoid let bindings, but I think using let would be clearer. You can achieve the same effect inside the `do` syntax with `let ... in` 
```suggestion
      let wdrl = TreasuryWithdrawals Map.empty SNothing
       in submitFailingGovAction
             wdrl
             [injectFailure $ ZeroTreasuryWithdrawals wdrl]
```
That's because:
```haskell
  let x = '5'
   in print x
  let x = '7'
   in print x
```
is different from
```haskell
  let y = 'a'
  print y
  let y = 'b' -- <- `y` name shadowing
  print y
```

Reply:
Much better!  I was trying to avoid `do` actually because I didn't want the ugly formatting with `do` at the beginning of the line. I like the  `let` version, thanks!